sale_commission_areamanager/models/account_invoice.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def write(self, vals):
+        result = super(AccountInvoice, self).write(vals)
+        for line in self.invoice_line_ids:
```

Comment:
Here you have to loop for invoice as follows:

```
for invoice in self:
    for line in invoice
```

otherwise when self is a recordset it raises an error as follows:

`ValueError: Expected singleton: account.invoice(4118, 3236)`